File: php/features.php
<?php
declare(strict_types=1);

// FreeJobs SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class FreeJobsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new FreeJobsBaseFeature();
            case "test":
                return new FreeJobsTestFeature();
            default:
                return new FreeJobsBaseFeature();
        }
    }
}

File: py/features.py
# FreeJobs SDK feature factory

from feature.base_feature import FreeJobsBaseFeature
from feature.test_feature import FreeJobsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FreeJobsBaseFeature(),
        "test": lambda: FreeJobsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
